src/mcp/client/auth/utils.py

```diff
@@ -243,6 +243,75 @@ async def handle_registration_response(response: Response) -> OAuthClientInforma
         raise OAuthRegistrationError(f"Invalid registration response: {e}")
 
 
+def is_valid_client_metadata_url(url: str | None) -> bool:
```

Comment:
open to a different break down of helper functions here, this seemed like a reasonably flexible combo.